backend/linux_input.c

```diff
@@ -116,22 +185,53 @@ static void *twin_linux_evdev_thread(void *arg)
         }
     }
 
-    /* Initialize pollfd array */
-    struct pollfd pfds[EVDEV_CNT_MAX];
-    for (int i = 0; i < evdev_cnt; i++) {
-        pfds[i].fd = evdev_fd[i];
+    /* Initialize evdev poll file descriptors  */
+    for (int i = EVDEV_IDX_START; i < evdev_cnt + UDEV_RESERVED_CNT; i++) {
+        pfds[i].fd = evdev_fd[i - 1];
         pfds[i].events = POLLIN;
     }
+}
+
+static void *twin_linux_evdev_thread(void *arg)
+{
+    twin_linux_input_t *tm = arg;
+
+    struct udev *udev = NULL;
+    struct udev_monitor *mon = NULL;
+
+    /* Open Linux udev (user space device manager) */
+    int udev_fd = twin_linux_udev_init(&udev, &mon);
+    if (udev_fd < 0) {
+        exit(2);
```

Comment:
The `exit` call here seems too harsh. Even if the input event thread fails, other parts of Mado should continue running - this would allow programmers to see the error messages and diagnose what went wrong. Consider handling the failure more gracefully rather than terminating the entire program.